global-script.lua

```diff
@@ -3890,3 +3896,34 @@ function onPlayerDisconnect(player)
         updatePlaySpiritButton(player.color)
     end
 end
+
+function onObjectSpawn(obj)
+    local is_power_card = obj.hasTag("Minor") or obj.hasTag("Major") or obj.hasTag("Unique")
+    if is_power_card then
+        applyPowerCardContextMenuItem(obj)
+    end
+end
+
+function applyPowerCardContextMenuItem(card)
+    card.addContextMenuItem(
+        "Forget",
+        function(player_color)
+            -- This ugliness is because setPositionSmooth doesn't work from a hand.
+            ensureCardInPlay(player_color, card)
+            discardPowerCardFromPlay(card, 1)
+        end,
+        false)
+end
+
+-- ensureCardInPlay moves the supplied card from the indicator player_color's
+-- hand to a safe location, if it's in any of their hands.
+function ensureCardInPlay(player_color, card)
```

Comment:
this function should return true or false depending upon if the card was found. since this code currently only checks the clicking player's hand (which is right imo). so if the card was in a different player's hand the position would never change, and which the function should return false. should also add a broadcast to the clicking player that that card is not in one of their hands

Reply:
If I understand correctly, you want the lambda passed to `addContextMenuItem` to return early if `ensureCardInPlay` returns false? This function would need to check all players' hands to do that, since not finding the card in the clicking player's hand doesn't imply that it's in a different player's hand. I want to also be able to forget from the table.

Reply:
well based on the comments it sounds like forget is not a safe operation to perform if the card is in someone's hand. and your code currently doesn't account for the case i try to forget a card in someone else's hand

Reply:
Personally, I'd prefer if it worked when a card is in any players hand. I often play multi-handed solo, or with trusted people, and don't feel like limiting what cards can be forgotten is a useful thing.

Reply:
I think I agree with Tom; inclined to do it that way.

Reply:
and @iakona, that is true. I think I glossed over it because it's "not safe" in that it doesn't do much of anything. But following Tom's suggestion would solve it too.

Reply:
hmm sure i'm fine with the code working on a card in anyone's hand. should be pretty obvious what is happening anyways. and someone can just as easily have moved the card manually if they wanted to be a bad actor, so this isn't really making things worse

Reply:
Now checks all `selectedColors`' hands.